packages/react-day-picker/src/contexts/SelectMultiple/SelectMultipleProvider.tsx

```diff
@@ -22,14 +21,23 @@ export function SelectMultipleProvider({
   initialProps: DayPickerProps;
   children: React.ReactNode;
 }): JSX.Element {
-  const [selectedDays, setSelectedDays] = useControllablePropState(
-    isDayPickerMultiple(initialProps)
-      ? {
-          value: initialProps.selected,
-          defaultValue: initialProps.defaultSelected
-        }
-      : { defaultValue: undefined }
-  );
+  const [selectedDays, setSelectedDays] = useControlledValue<
+    Date[] | undefined
```

Comment:
I think I might know a way to fix this. We can’t conditionally call a hook, but we can conditionally call a subcomponent for which typescript is confident about the narrowing type checked by isDayPickerMultiple. At that point the values of the parameters to useControlledValue will be known to typescript